File: common/src/main/java/dev/obscuria/fragmentum/client/FragmentumClientRegistry.java
package dev.obscuria.fragmentum.client;

import dev.obscuria.fragmentum.Fragmentum;
import net.minecraft.client.gui.screens.inventory.tooltip.ClientTooltipComponent;
import net.minecraft.world.inventory.tooltip.TooltipComponent;

import java.util.function.Function;

@SuppressWarnings("unused")
public interface FragmentumClientRegistry
{
    static ClientRegistrar registrar(String modId)
    {
        return Fragmentum.SERVICES.client().registrar(modId);
    }

    static <T extends TooltipComponent> void registerTooltipComponent(Class<T> type, Function<T, ClientTooltipComponent> factory)
    {
        Fragmentum.SERVICES.client().registerTooltipComponent(type, factory);
    }
}

File: common/src/main/kotlin/dev/obscuria/fragmentum/client/FragmentumClientRegistry.kt
package dev.obscuria.fragmentum.client

import dev.obscuria.fragmentum.Fragmentum
import net.minecraft.client.gui.screens.inventory.tooltip.ClientTooltipComponent
import net.minecraft.world.inventory.tooltip.TooltipComponent
import java.util.function.Function

object FragmentumClientRegistry {

    fun registrar(modId: String): ClientRegistrar {
        return Fragmentum.SERVICES.client().registrar(modId)
    }

    fun <T : TooltipComponent> registerTooltipComponent(
        type: Class<T>,
        factory: Function<T, ClientTooltipComponent>
    ) {
        Fragmentum.SERVICES.client().registerTooltipComponent(type, factory)
    }
}